src/IonTextReader.ts

```diff
@@ -92,8 +92,23 @@ export class TextReader implements Reader {
         if(this.depth() > 0) return false;
         if(input.length < 6 || this.annotations().length > 0) return false;
         let prefix = "$ion_";
-        for(let i = 0; i < prefix.length; i++){
+        let i = 0;
+
+        while(i < prefix.length){
             if(prefix.charAt(i) !== input.charAt(i)) return false;
+            i++;
+        }
+        let tempChar;
+        while(input.charAt(i) != '_'){
```

Comment:
potential index out of bounds issue here?

Reply:
it returns undefined which triggers the false case.

Reply:
looks like `"hello".charAt(1000)` returns an empty string (different than `undefined`), but the effect is the same.  ok